src/main/kotlin/config/context/listener/ResponseListener.kt

```diff
@@ -0,0 +1,12 @@
+package config.context.listener
+
+import config.context.sessionContext
+
+class ResponseListener: EventListener {
+  private val authUserHeaderName = "AuthUser"
+
+  override fun update(eventType: EventTypes) {
+    val newAuthUser: String? = sessionContext().serviceResponse?.getHeaderByName(authUserHeaderName)
```

Comment:
sessionContext().apply {
    serviceResponse?.getHeaderByName(authUserHeaderName)?.also { authUser = it }
}